### PR TITLE
Sergiu uncu changes

### PR DESCRIPTION
**Pull request by marcghelas**
-merge google pay changes with Android-Sample-App
-add RxJava dependency for Gpay payment
-add Google Pay Services dependency

### CODE REVIEW

File: samples/src/main/java/com/judopay/samples/MainActivity.java [hunk 1]
```diff
@@ -234,4 +334,36 @@ private void handleResult(int resultCode, Intent data) {
...
+        Judo judo = getJudo();
+        PaymentsClient paymentClient = GooglePaymentUtils.getGooglePayPaymentsClient(this, WalletConstants.ENVIRONMENT_PRODUCTION);
+
+        GooglePaymentUtils.checkIsReadyGooglePay(paymentClient, result -> {
+            if (result) {
+                tvGPayNotSupport.setVisibility(View.GONE);
+                btGpay.setVisibility(View.VISIBLE);
+
+                PaymentDataRequest paymentDataRequest = GooglePaymentUtils.createDefaultPaymentDataRequest(judo);
+                final Task<PaymentData> taskDefaultPaymentData = paymentClient.loadPaymentData(paymentDataRequest);
```
Comment (zfoltin) [inline samples/src/main/java/com/judopay/samples/MainActivity.java]: `paymentData`

File: samples/src/main/java/com/judopay/samples/MainActivity.java [hunk 2]
```diff
@@ -234,4 +334,36 @@ private void handleResult(int resultCode, Intent data) {
                 break;
         }
     }
+
+    private void setUpGpay() {
+        Judo judo = getJudo();
+        PaymentsClient paymentClient = GooglePaymentUtils.getGooglePayPaymentsClient(this, WalletConstants.ENVIRONMENT_PRODUCTION);
```
Comment (zfoltin) [inline samples/src/main/java/com/judopay/samples/MainActivity.java]: `paymentsClient`

File: samples/build.gradle [hunk 3]
```diff
@@ -36,4 +36,14 @@ dependencies {
 
     testImplementation 'junit:junit:4.12'
     testImplementation 'org.mockito:mockito-core:2.28.2'
+
+//    Google Pay Services
+    api('com.google.android.gms:play-services-wallet:16.0.1') {
```
[the remark this answers is not in the document]
Comment (zfoltin) [reply samples/build.gradle]: Any reason why these are `api` instead of `implementation`?

**Comment by marcghelas** [comment]
@zfoltin hi Zeno.
These are the changes from the previous developer, haven't touched anything and just made a PR. 
Should have had a look first! sorry for that

File: samples/src/main/java/com/judopay/samples/MainActivity.java (hunk 2 above)
Comment (marcghelas) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: changed

File: samples/src/main/java/com/judopay/samples/MainActivity.java (hunk 1 above)
Comment (marcghelas) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: changed

File: samples/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -109,5 +124,27 @@
...
+                android:layout_height="wrap_content"
+                android:layout_gravity="center"
+                android:text="Device not supprot GPAY" />
+
+            <include
+                android:id="@+id/btnGPay"
+                layout="@layout/googlepay_button_no_shadow"
+                android:layout_width="match_parent"
+                android:layout_height="48dp"
+                android:layout_margin="16dp"
```
[the remark this answers is not in the document]
Comment (marcghelas) [reply samples/src/main/res/layout/activity_main.xml]: extracted

File: samples/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -109,5 +124,27 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical">
+
+            <TextView
+                android:id="@+id/tvGPayNotSupport"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center"
+                android:text="Device not supprot GPAY" />
```
[the remark this answers is not in the document]
Comment (marcghelas) [reply samples/src/main/res/layout/activity_main.xml]: I will make it's visibility GONE for now it that's ok
string extracted

File: samples/src/main/java/com/judopay/samples/MainActivity.java [hunk 6]
```diff
@@ -30,16 +56,35 @@
...
 public class MainActivity extends BaseActivity {
+    private static final String TAG = MainActivity.class.getSimpleName();
     private static final String JUDO_ID = "<JUDO_ID>";
     private static final String API_TOKEN = "<API_TOKEN>";
     private static final String API_SECRET = "<API_SECRET>";
     private static final String AMOUNT = "0.10";
     private static final String REFERENCE = UUID.randomUUID().toString();
 
+    private TextView tvGPayNotSupport;
+    private View btGpay;
```
[the remark this answers is not in the document]
Comment (marcghelas) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: changed

File: samples/src/main/java/com/judopay/samples/MainActivity.java [hunk 7]
```diff
@@ -30,16 +56,35 @@
...
 
 public class MainActivity extends BaseActivity {
+    private static final String TAG = MainActivity.class.getSimpleName();
     private static final String JUDO_ID = "<JUDO_ID>";
     private static final String API_TOKEN = "<API_TOKEN>";
     private static final String API_SECRET = "<API_SECRET>";
     private static final String AMOUNT = "0.10";
     private static final String REFERENCE = UUID.randomUUID().toString();
 
+    private TextView tvGPayNotSupport;
```
[the remark this answers is not in the document]
Comment (marcghelas) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: changed

File: samples/build.gradle [hunk 8]
```diff
@@ -36,4 +36,14 @@ dependencies {
...
+        exclude group: 'com.android.support', module: 'support-media-compat'
+        exclude group: 'com.android.support', module: 'support-v4'
+    }
+
+//    RxJava
+    api 'io.reactivex.rxjava2:rxjava:2.2.10'
```
[the remark this answers is not in the document]
Comment (marcghelas) [reply samples/build.gradle]: Raised

File: samples/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -109,5 +124,28 @@
             android:text="@string/test_card_detail_instructions"
             android:textSize="14sp"
```
Comment (stpopa) [inline samples/src/main/res/layout/activity_main.xml]: let's extract "14sp" to resources please

Comment (marcghelas) [reply samples/src/main/res/layout/activity_main.xml]: thanks, extracted

**Comment by marcghelas** [comment]
@zfoltin @stpopa Is it okay if I will write new files in Kotlin for future changes? Or keep it consistent and write java only?

**Comment by zfoltin** [comment]
> @zfoltin @stpopa Is it okay if I will write new files in Kotlin for future changes? Or keep it consistent and write java only?

I'd rather we don't add the Kotlin dependency for now and keep it Java only.

File: samples/build.gradle [hunk 10]
```diff
@@ -36,4 +36,14 @@ dependencies {
...
     testImplementation 'junit:junit:4.12'
     testImplementation 'org.mockito:mockito-core:2.28.2'
+
+//    Google Pay Services
+    implementation('com.google.android.gms:play-services-wallet:18.0.0') {
+        exclude group: 'com.android.support', module: 'support-media-compat'
+        exclude group: 'com.android.support', module: 'support-v4'
+    }
+
+//    RxJava
```
Comment (zfoltin) [inline samples/build.gradle]: I missed this in the previous round, sorry, but can we remove these comments (they are meaningless) and move the `implementation` things up to line 35 (above the `testImplementation`s)?

File: samples/src/main/java/com/judopay/samples/MainActivity.java (hunk 1 above)
Comment (zfoltin) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: no, you haven't :)

Comment (zfoltin) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: not changed yet :)

Comment (marcghelas) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: Did you mean to change variable name paymentDataRequest to paymentData or was it something else?

File: samples/build.gradle (hunk 10 above)
Comment (marcghelas) [reply samples/build.gradle]: sure. Done

File: samples/src/main/java/com/judopay/samples/MainActivity.java (hunk 1 above)
Comment (zfoltin) [reply samples/src/main/java/com/judopay/samples/MainActivity.java]: It was to rename `taskDefaultPaymentData` to `paymentData` - not a big issue, it's just that the "taskDefault" part of the variable name makes very little sense

**Pull request merged**